experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,202 @@
+# I would like to credit Dario Mathiä who produced an initial version of the following code
+
+export wreath_macs, fake_deg
+
+Oscar.canonical_unit(x::QQAbFieldElem{AbsSimpleNumFieldElem}) = x #meanwhile canonical_unit gets fixed
+
+# Tools
+
+#gives indices of a subarray
+function subarray_indices(sub,arr)
+  return [findfirst(==(s), arr) for s in sub]
+end
+
+# Computes the b-invariant
+function b1(lambda::Vector{Int})
+  return sum((i-1)*l_i for (i, l_i) in enumerate(lambda); init=0)
+end
+
+# Computes the b-invariant of a multipartition
+function b_inv(lbb::Multipartition)
+  return b1(map(lambda -> sum(lambda)[1], lbb)) + length(lbb)*sum(i -> b1(data(lbb[i])),1:length(lbb))
+end
+
+function beta_number_to_partition(beta::Vector{Int})
+  lb=Int[]
+  lbeta=length(beta)
+  for j in 1:lbeta
+    nbholes=beta[j]-beta[1]-(j-1)
+    if nbholes >= 1
+    pushfirst!(lb, nbholes)
+    end
+  end
+  return partition(lb)
+end
+
+# Residue notation is in terms of rows and columns
+
+function core(coroot::Vector{Int},r::Int)
+  beta=Int[]
+  m=minimum(coroot)
+  M=maximum(coroot)
+  for k in m:M
+    for j in 1:r
+       if k <= coroot[j]
+          push!(beta, k*r+(j-1))
+       end
+    end
+  end
```

Comment:
This is a matter of taste; but if wanted, you could turn this into a single list comprehension:
```suggestion
  m=minimum(coroot)
  M=maximum(coroot)
  beta = [k*r+(j-1) for k in m:M for j in 1:r if k <= coroot[j]
```

Reply:
I will do that, it is way shorter. Thanks